src/main/java/org/jabref/gui/dialogs/BackupUIManager.java

```diff
@@ -91,13 +99,21 @@ private static Optional<ParserResult> showReviewBackupDialog(
                         originalDatabase, "Review Backup"
                 );
                 var allChangesResolved = dialogService.showCustomDialogAndWait(reviewBackupDialog);
-                if (allChangesResolved.isEmpty() || !allChangesResolved.get()) {
-                    // In case not all changes are resolved, start from scratch
-                    return showRestoreBackupDialog(dialogService, originalPath, preferencesService, fileUpdateMonitor);
+                LibraryTab saveState = stateManager.activeTabProperty().get().get();
+                changes.stream().filter(DatabaseChange::isAccepted).forEach(change -> change.applyChange(ce));
+                ce.end();
```

Comment:
Add the CE to the undomanager

```java
undoManager.addEdit(ce);
```

Initialize the ce right before line 103. This variable is more local.